bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/AuthHandler.java

```diff
@@ -92,6 +96,24 @@ public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception
             }
 
             if (authenticated) {
+                if (msg instanceof BookkeeperProtocol.Request) {
```

Comment:
@eolivelli 
I can make it use EUA as opposed to creating a new Error code.
However, just using EUA won't work right? The check for older Versions fails here because it tests for message to be an instance of latest version - V3 and it seems incompatible with V2.

If anything I feel we need to add another check here.
If we are in here, we know that the request is already authenticated, so we can cast `msg` to `BookkeeperProtocol.Request` and do a `.getVersion()` on that. If that works, it seems that the message is of type Request (may even be an earlier version)

Something like this:

```
           if (authenticated) {
                if (((BookieProtocol.Request) msg).getProtocolVersion() || msg instanceof BookkeeperProtocol.Request) {
                    Channel c = ctx.channel();
```

Reply:
@Ghatage 
this is probably the wrong place for this check.
If you allow ONLY secured client you must forbid the auth as well, because the risk is to exchange credentials without encryption.
I didn't check well, but I hope that startTLS happens before exchanging auth

so your check should be:
- in `if (msg instanceof BookieProtocol.AuthRequest) { // pre-PB-client` block
- in `if (msg instanceof BookkeeperProtocol.Request) { // post-PB-client` block

The former is v2 protocol, and you should simply fail the request, because there is no support for TLS
The latter is v3 protocol and you should your check `c.pipeline().get(SslHandler.class) == null`

O hope that helps